### PR TITLE
Week 5 tuan le

### PR DESCRIPTION
**Pull request by TuanLeVietz**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [X] JavaScript Coding Challenges
- [X] Ruby Coding Challenges
- [X] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -21,7 +21,26 @@ const secretCodeWord2 = "Gobbledygook"
...
+})
+
 // b) Create the function that makes the test pass.
+/*Pseudocode: Create a function called secretCode that takes in a string as its argument. We then create a variable word1 thats gonna equal the string and use .replace to replace the given words to the given numbers within the string(remember to also include the capital E for the 3d word) . 
+*/
+
+const secretCode = (string) => {
+    var word1 = string.replace(/a/g, 4).replace(/e/g, 3).replace(/i/g, 1).replace(/o/g, 0).replace(/E/g, 3)
+        return word1
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:42]: Replace is a cool string method huh! Love what you did with the capital 'E' and I want you to do that for all the different letters that might come your way.

File: code-challenges.test.js [hunk 2]
```diff
@@ -34,9 +53,27 @@ const letterA = "a"
...
+
+
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+/*Pseudocode: create a high order function named letterContainer that takes in array and letter as its arguments. We then return the array with .filter to filter the words that include a or e. Since every word is capitalized we use .includes twice, one regular and the other one with .toUpperCase, divided by the OR boolean.*/
+
+const letterContainer = (array, letter) => {
+    return array.filter(value => value.includes(letter) || value.includes(letter.toUpperCase()))
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:73]: Good work!

File: code-challenges.test.js [hunk 3]
```diff
@@ -49,4 +86,31 @@ const hand3 = [5, 5, 5, 5, 4]
...
+    let check = array.sort()
+    for (let i = 0;i < check.length; i++){
+        if(check[0] === check[1] && check[1] === check[2] && check[3] === check[4]){
+            return true
+        }else if(check[0] === check[1] && check[2] === check[3] && check[2] === check[4]){
+            return true
+        }else 
+        return false
+    }
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:115]: Well done! What would we do if we included suits (like an emoji)?

File: ruby-challenges.rb [hunk 4]
```diff
@@ -10,23 +10,71 @@
...
+        p "The #{model}'s bike is so fast with its #{wheels} burning wheels that when it's going #{current_speed} mph, it leaves a trail of fire!"
+    end
+    def pedal_faster current_speed
+        @current_speed += current_speed
+    end
+    def brake current_speed
+        @current_speed -= current_speed
+        [@current_speed, 0].max 
+    end
+end
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:65]: Very cool! notcied this solution on another assessment too. I haven't seen it before this.

**Review by WalkerTexas-coder — APPROVED** [review]
Good job Tuan! You are on the right path! Remember your test variables are there to help guide you to create code for all similar conditions not just the test conditions provided. Often as devs we'll have limited testing but our code has to account for things called edge cases. Besure to incorporate edge cases to your problem solving. Good work! Approved!
